ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/utils/ThreadUtil.java

```diff
@@ -28,7 +31,7 @@
     /**
      * The ID associated with the thread.
      */
-    private static final ThreadLocal THREAD_ID = new ThreadLocal() ;
+    private static final Map<Thread,String> THREAD_ID = new HashMap<Thread,String>() ;
```

Comment:
I meant use a long inside the ThreadUtil class but convert them to Strings in the interface.